imap_processing/tests/lo/test_science_direct_events.py

```diff
@@ -37,294 +55,149 @@ def tof_data():
+    tof2 = "000000010"  # 2
+    tof3 = "000011"  # 3
+    cksm = "000"  # 0
+    # POS not trasmitted
+    single_de.DATA = absent + time + energy + mode + tof0 + tof2 + tof3 + cksm
+    data = BinaryString(single_de.DATA)
+
+    expected_time = np.array([100])
+    expected_energy = np.array([2])
+    expected_mode = np.array([1])
+    # tofs and cksm are bit shifted to the left by 1 during decompression
+    expected_tof0 = np.array([0 << 1])
+    expected_tof1 = np.array([GlobalConstants.DOUBLE_FILLVAL])
+    expected_tof2 = np.array([2 << 1])
+    expected_tof3 = np.array([3 << 1])
+    expected_cksm = np.array([0 << 1])
+    expected_pos = np.array([GlobalConstants.DOUBLE_FILLVAL])
 
     # Act
-    tof_calc_bin = single_de._read_tof_calculation_table(case_number)
+    single_de._parse_data(data)
 
     # Assert
-    assert tof_calc_bin == binary_strings_expected
-
-
-@pytest.mark.skip(reason="no data to initialize with")
-def test_find_remaining_bits(single_de):
+    np.testing.assert_array_equal(single_de.TIME, expected_time)
+    np.testing.assert_array_equal(single_de.ENERGY, expected_energy)
+    np.testing.assert_array_equal(single_de.MODE, expected_mode)
+    np.testing.assert_array_equal(single_de.TOF0, expected_tof0)
+    np.testing.assert_array_equal(single_de.TOF1, expected_tof1)
+    np.testing.assert_array_equal(single_de.TOF2, expected_tof2)
+    np.testing.assert_array_equal(single_de.TOF3, expected_tof3)
+    np.testing.assert_array_equal(single_de.CKSM, expected_cksm)
+    np.testing.assert_array_equal(single_de.POS, expected_pos)
+
+
+def test_parse_data_case_10(single_de):
     # Arrange
-    single_de.DATA = "000100010101"
-    bitstream = bitstring.ConstBitStream(bin=single_de.DATA)
-    case_number = single_de._find_decompression_case(bitstream)
-    tof_calc = single_de._read_tof_calculation_table(case_number)
-    remaining_coeff_expected = {
-        "TIME": [
-            327.68,
-            163.84,
-            81.82,
-            40.96,
-            20.48,
-            10.24,
-            5.12,
-            2.56,
-            1.28,
-            0.64,
-            0.32,
-            0.16,
-        ],
-        "ENERGY": [
-            0.32,
-            0.16,
-        ],
-        "TOF0": [
-            163.84,
-            81.82,
-            40.96,
-            20.48,
-            10.24,
-            5.12,
-            2.56,
-            1.28,
-            0.64,
-            0.32,
-        ],
-        "TOF1": [
-            81.82,
-            40.96,
-            20.48,
-            10.24,
-            5.12,
-            2.56,
-            1.28,
-            0.64,
-            0.32,
-        ],
-        "TOF2": [
-            81.82,
-            40.96,
-            20.48,
-            10.24,
-            5.12,
-            2.56,
-            1.28,
-            0.64,
-            0.32,
-        ],
-        "TOF3": [],
-        "POS": [],
-        "CKSM": [],
-    }
+    absent = "1010"  # case 10
```

Comment:
This is a really nice way to do this testing :) Well done!

Reply:
agree!